### ISSUE TITLE
Need to generate tfidf vector probably 50 dimension for the big whole event dataset.

### ISSUE THREAD

**Issue by YanLiang1102** [oudalab/co-incidents#9]
using parallel that Dr. Grant suggest and store the data in sqlite, runtime either implement cache and also look into the database to get it , no need to load all of them in memory.

**Comment by YanLiang1102** [comment]
'''
['000', '10', '100', '11', '12', '13', '14', '15', '18', '20', '25', '30', '50', 'according', 'action', 'added', 'administration', 'afghan', 'afghanistan', 'agency', 'ago', 'agreement', 'air', 'al', 'american', 'announced', 'april', 'area', 'army', 'asked', 'attack', 'attacks', 'authorities', 'bank', 'based', 'billion', 'board', 'border', 'british', 'bush', 'business', 'called', 'came', 'campaign', 'canada', 'canadian', 'capital', 'case', 'cent', 'central', 'change', 'chief', 'children', 'china', 'chinese', 'city', 'come', 'commission', 'committee', 'community', 'companies', 'company', 'conference', 'continue', 'control', 'council', 'countries', 'country', 'court', 'day', 'days', 'deal', 'death', 'decision', 'defence', 'department', 'development', 'did', 'director', 'district', 'does', 'don', 'earlier', 'early', 'east', 'economic', 'economy', 'election', 'end', 'energy', 'english', 'european', 'executive', 'expected', 'family', 'far', 'federal', 'financial', 'force', 'forces', 'foreign', 'free', 'friday', 'future', 'gas', 'general', 'given', 'going', 'good', 'government', 'group', 'groups', 'head', 'health', 'held', 'help', 'high', 'home', 'hospital', 'house', 'human', 'including', 'increase', 'industry', 'information', 'international', 'iran', 'iranian', 'iraq', 'islamic', 'israel', 'israeli', 'issue', 'issues', 'john', 'judge', 'just', 'justice', 'killed', 'know', 'known', 'later', 'law', 'leader', 'leaders', 'led', 'left', 'life', 'like', 'line', 'little', 'local', 'london', 'long', 'major', 'make', 'making', 'man', 'march', 'market', 'media', 'meeting', 'member', 'members', 'men', 'military', 'million', 'minister', 'ministry', 'monday', 'money', 'month', 'months', 'mr', 'ms', 'national', 'nations', 'near', 'need', 'new', 'news', 'night', 'north', 'nuclear', 'number', 'obama', 'office', 'official', 'officials', 'oil', 'old', 'ontario', 'opposition', 'order', 'ottawa', 'pakistan', 'palestinian', 'parliament', 'party', 'past', 'pay', 'peace', 'people', 'place', 'plan', 'plans', 'police', 'policy', 'political', 'power', 'president', 'press', 'price', 'prime', 'private', 'process', 'program', 'province', 'provincial', 'public', 'recent', 'region', 'relations', 'report', 'reported', 'reports', 'right', 'rights', 'run', 'russia', 'russian', 'said', 'say', 'saying', 'says', 'school', 'second', 'secretary', 'security', 'senior', 'service', 'services', 'set', 'shows', 'soldiers', 'source', 'south', 'special', 'spokesman', 'staff', 'state', 'statement', 'states', 'students', 'support', 'syria', 'syrian', 'taken', 'talks', 'tax', 'team', 'think', 'thursday', 'time', 'times', 'today', 'told', 'took', 'toronto', 'trade', 'troops', 'tuesday', 'tv', 'union', 'united', 'university', 'use', 'used', 'video', 'visit', 'want', 'war', 'washington', 'way', 'website', 'wednesday', 'week', 'west', 'western', 'white', 'women', 'won', 'work', 'workers', 'working', 'world', 'year', 'years', 'yesterday', 'york']
'''
this is the keyword for 300 dimension feature_names I have well is 50 dimension really enough?

**Comment by YanLiang1102** [comment]
so i can control the max_df and min_df, think of use some way to choose the best max_df and min_df, how to exclude some word that I don't want in the feature names??? something to think about...
like those digit I really don't want them @cegme

**Comment by cegme** [comment]
You could plot the distribution (words x df) to choose a cutoff point on the long tail. You can be aggressive early on and find the best point when you start optimizing.

**Comment by YanLiang1102** [comment]
@cegme  yeah I got what you mean this time, that is a good idea I think u tole me that before I don't knwo what exactly u mean at that time. for each words visualize its doc frequency and make the max_df and min_df based on that, so right now the vector I have is not tfidf when I use sklearn to train it got error, so it is just one hot encoding, so say if in the document "like" is the feature word , and it appears twice in that doc it will be "2" as the component in the vector, but if I choose good max_df, and min_df well, this one-hot should be similar to tfidf. I Think
